gridded_etl_tools/utils/store.py

```diff
@@ -280,6 +305,25 @@ def get_metadata_path(self, title: str, stac_type: str) -> str:
         else:
             return f"s3://{self.bucket}/metadata/{title}.json"
 
+    def write_metadata_only(self, update_attrs: dict[str, Any]):
+        # Edit both .zmetadata and .zattrs
+        for z_path in (".zmetadata", ".zattrs"):
+            current_attributes = {}
+
+            # Read current metadata from Zarr
+            with self.fs().open(f"{self.path}/{z_path}") as z_contents:
+                current_attributes.update(json.load(z_contents))
+
+            # Update given attributes at the appropriate location depending on which z file
+            if z_path == ".zmetadata":
+                current_attributes["metadata"][".zattrs"].update(update_attrs)
+            else:
+                current_attributes.update(update_attrs)
+
+            # Write back to Zarr
+            with self.fs().open(f"{self.path}/{z_path}", "w") as z_contents:
+                json.dump(current_attributes, z_contents)
```

Comment:
There's no need to instantiate a blank dict at the start of the for loop. Just set `current_attributes = json.load(z_contents)` on line 315,